add-on/chrome/test/mochitest/browser_LoopSidebar.js

```diff
@@ -0,0 +1,52 @@
+/* Any copyright is dedicated to the Public Domain.
+   http://creativecommons.org/publicdomain/zero/1.0/ */
+
+/*
+ * This file contains tests for the LoopSidebar
+ */
+"use strict";
+
+const { LoopSidebar } = Cu.import("chrome://loop/content/modules/LoopSidebar.jsm", {});
+
+var win;
+
+registerCleanupFunction(function* () {
+  if (win) {
+    yield BrowserTestUtils.closeWindow(win);
+  }
+});
+
+function goOffline() {
+  // Turn off the network for loop tests, so that we don't
+  // try to access the remote servers. If we want to turn this
+  // back on in future, be careful to check for intermittent
+  // failures.
+  Services.io.offline = true;
+
+  registerCleanupFunction(function() {
+    Services.io.offline = WAS_OFFLINE;
+  });
+}
+
+add_task(function* test_LoopSidebar_noSidebar() {
+  let sidebar = window.document.getElementById("loop-side-browser");
+  Assert.ok(!sidebar, "there should be no sidebars yet");
+});
+
+add_task(function* test_LoopSidebar_createSidebar() {
+  goOffline();
+  win = yield BrowserTestUtils.openNewBrowserWindow();
+  LoopSidebar.createSidebar(win, "faketoken");
+  let sidebar = win.document.getElementById("loop-side-browser");
+  Assert.ok(sidebar, "there should create a sidebar");
+  Assert.strictEqual(sidebar.getAttribute("src"), "about:loopconversation#faketoken", "Url must be set properly");
+});
+
+add_task(function* test_LoopSidebar_checkPermissions() {
+  goOffline();
+  win = yield BrowserTestUtils.openNewBrowserWindow();
+  LoopSidebar.createSidebar(win, "faketoken");
+  Assert.equal(Services.perms.testPermission("about:loopconversation#faketoken", "camera"),
+    Services.perms.ALLOW_ACTION, "Camera permissions should be granted");
+});
```

Comment:
@Standard8 I'm getting an error when running the last test:

> 25 INFO TEST-UNEXPECTED-FAIL | browser/extensions/loop/chrome/test/mochitest/browser_LoopSidebar.js | Uncaught exception - [Exception... "Could not convert JavaScript argument arg 0 [nsIPermissionManager.testPermission]"  nsresult: "0x80570009 (NS_ERROR_XPC_BAD_CONVERT_JS)"  location: "JS frame :: chrome://mochitests/content/browser/browser/extensions/loop/chrome/test/mochitest/browser_LoopSidebar.js :: test_LoopSidebar_checkPermissions :: line 49"  data: no`

It seems that complains about the result of testPermission but according to mdn documentation, that function returns a boolean